### ISSUE TITLE
Bug: System.AccessViolationException on IntermediateDecode from .NET

### ISSUE THREAD

**Issue by Henderz** [coqui-ai/STT#1952]
**Describe the bug**
(As I understand the native_client for .Net is still 'DeepSpeech'; Please correct me if I am wrong.)
I am using DeepSpeech for inference from microphone stream captured via CSCore audio module. I have custom code for VAD and get Intermediate decoding done to get sentence wise live transcription.

Models: 9.0.3 Pre-Trained English Audio Model and custom Scorers with the same hyper-parameters as the Pre-Trained Scorer.

This works but at random times I get the following "unhandled" exceptions from the .so file.
StackOverflowException
from objDeepSpeech.FeedAudioContent(objStream, buffers, Convert.ToUInt32(buffers.Length));
or
System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'
from objDeepSpeech.IntermediateDecodeWithMetadata(objStream, 1);

The errors originate from within libdeepspeech.so, so I am not able to debug any further. Any help is much appreciated. Thanks.

**To Reproduce**
1) Use CSCore to listen to Mic
2) Call objDeepSpeech.IntermediateDecodeWithMetadata(objStream, 1); occasionally - (VAD or just randomly)
3) Between 30 secs to 2 mins, the call to decode fails with AccessViolationException and is not handle-able as its from the unmanaged c[++] compiled code from the .so file.

**Expected behavior**
1) For the calls to keep working indefinitely as long as the system resources allow it. 

**Environment (please complete the following information):**
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Windows 10
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: NA
- **TensorFlow version (use command below)**: NA
- **Python version**: NA
- **Bazel version (if compiling from source)**: NA
- **GCC/Compiler version (if compiling from source)**: NA
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NA
- **Exact command to reproduce**: objDeepSpeech.IntermediateDecodeWithMetadata(objStream, 1);

**Additional context**
No failure is seen when using NAudio but STT accuracy is a lot worser than when using CSCore for the same audio. 

Can NAudio be made better instead for Intermediate Decoding?
Is any other language implicitly better at this than C#?

**Comment by reuben** [comment]
Can you provide a script/repo that can be built/run to reproduce this? Ideally without requiring listening from a live mic. And also the full error stack (including the native frames if that's possible). Thank you!
